src/tracing/esp32_trace/include/matter/tracing/macros_impl.cpp

```diff
@@ -28,17 +38,47 @@ namespace Insights {
                        heap_caps_get_free_size(MALLOC_CAP_8BIT));                                                                  \
     } while (0)
 
+void ESP32Backend::InitializeBlacklistHashes()
+{
+    for (int i = 0; i < blacklistSize; i++)
+    {
+        if (blackListStrings[i] == NULL)
+            break;
+        size_t data_length = strlen(blackListStrings[i]);
```

Comment:
a function of `uint64_t hashString(const char *string)` should be split out as it seems shared between both initialize and lookup. 

How often is this method assumed to be called? SHA1 seems quite resource intensive (e.g. a murmurhash may be 10x as fast and actually create a small hash value).